### PR TITLE
WIP: Use the latest release of AllenNLP

### PR DESCRIPTION
**Pull request by schmmd**
The following models need to be addressed:

- [x] Semantic Role Labeling
- [x] NER
- [x] Fine-grained NER
- [x] Constituency parser
- [x] Dependency parser
- [x] Open IE
- [x] Sentiment Analysis
- [x] Coreference Resolution
- [x] BiDAF
- [x] ELMo-BiDAF
- [x] NAQANet
- [x] Semantic Parsing
- [x] Language Modeling
- [x] Masked Language Modeling

### CODE REVIEW

**Comment by ZhaofengWu** [comment]
Once this is done, we should be able to removed the specific files for coref, e.g. `Dockerfile.coref`, right?

**Comment by matt-gardner** [comment]
@ZhaofengWu, we'll still need one for a new reading comprehension demo that's coming soon, that depends on a third-party library.

**Comment by schmmd** [comment]
@ZhaofengWu yes, we won't need it for coref but Matt is right--we'll need something similar for another model.

File: models.json [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
      },
     "naqanet-reading-comprehension": {
-        "archive_file": "https://storage.googleapis.com/allennlp-public-models/naqanet-2019.04.29-fixed-weight-names.tar.gz",
-        "predictor_name": "machine-comprehension",
+        "archive_file": "https://storage.googleapis.com/allennlp-public-models/naqanet-2020.02.19.tar.gz",
+        "predictor_name": "reading-comprehension",
         "max_request_length": 25819
     },
     "semantic-role-labeling": {
-        "archive_file": "https://storage.googleapis.com/allennlp-public-models/bert-base-srl-2019.06.17.tar.gz",
```
Comment (DeNeutoy) [inline models.json]: Please could we keep the bert-base part of this filename?

Comment (schmmd) [reply models.json]: Yep--looks like I grabbed the wrong model.

Comment (schmmd) [reply models.json]: Yes-looks like I accidentally chose the wrong image.

Comment (schmmd) [reply models.json]: This is resolved.

**Comment by schmmd** [comment]
At this point, everything works except the interpret functionality on the language model.

**Comment by matt-gardner** [comment]
@schmmd, do you need me to look at something for that?

**Comment by schmmd** [comment]
Not presently, thanks. I'm going to take a closer look first.

**Comment by schmmd** [comment]
@matt-gardner looks like there's another problem in the model itself.  I'm going to revert it in my PR and use an older version for now.

```
{"message": "[2020-02-24 21:23:23,140] ERROR in app: Exception on /interpret/next-token-lm [POST]   
Traceback (most recent call last):                                                                  
  File \"/usr/local/lib/python3.6/site-packages/flask/app.py\", line 2446, in wsgi_app              
    response = self.full_dispatch_request()                                                         
  File \"/usr/local/lib/python3.6/site-packages/flask/app.py\", line 1951, in full_dispatch_request 
    rv = self.handle_user_exception(e)                                                              
  File \"/usr/local/lib/python3.6/site-packages/flask_cors/extension.py\", line 161, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))                                
  File \"/usr/local/lib/python3.6/site-packages/flask/app.py\", line 1820, in handle_user_exception 
    reraise(exc_type, exc_value, tb)                                                                
  File \"/usr/local/lib/python3.6/site-packages/flask/_compat.py\", line 39, in reraise             
    raise value                                                                                     
  File \"/usr/local/lib/python3.6/site-packages/flask/app.py\", line 1949, in full_dispatch_request 
    rv = self.dispatch_request()                                                                    
  File \"/usr/local/lib/python3.6/site-packages/flask/app.py\", line 1935, in dispatch_request      
    return self.view_functions[rule.endpoint](**req.view_args)                                      
  File \"./app.py\", line 436, in interpret                                                         
    interpretation = _caching_interpret(interpreter, json.dumps(data))                              
  File \"./app.py\", line 179, in _caching_interpret                                                
    return interpreter.saliency_interpret_from_json(json.loads(data))                               
  File \"/usr/local/lib/python3.6/site-packages/allennlp/interpret/saliency_interpreters/integrated_gradient.py\", line 20, in saliency_interpret_from_json
    labeled_instances = self.predictor.json_to_labeled_instances(inputs)                            
  File \"/usr/local/lib/python3.6/site-packages/allennlp/predictors/predictor.py\", line 80, in json_to_labeled_instances
    new_instances = self.predictions_to_labeled_instances(instance, outputs)                        
  File \"/usr/local/lib/python3.6/site-packages/allennlp/predictors/next_token_lm.py\", line 29, in predictions_to_labeled_instances
    vocab=self._model.vocab,                                                                        
  File \"/usr/local/lib/python3.6/site-packages/allennlp/data/instance.py\", line 50, in add_field  
    field.index(vocab)                                                                              
  File \"/usr/local/lib/python3.6/site-packages/allennlp/data/fields/text_field.py\", line 65, in index
    self._indexed_tokens[indexer_name] = indexer.tokens_to_indices(self.tokens, vocab)              
  File \"/usr/local/lib/python3.6/site-packages/allennlp/data/token_indexers/pretrained_transformer_indexer.py\", line 99, in tokens_to_indices
    indices, type_ids = self._extract_token_and_type_ids(tokens)                                    
  File \"/usr/local/lib/python3.6/site-packages/allennlp/data/token_indexers/pretrained_transformer_indexer.py\", line 124, in _extract_token_and_type_ids
    \"Using PretrainedTransformerIndexer but field text_id is not set\"                             
KeyError: 'Using PretrainedTransformerIndexer but field text_id is not set for the following token: \u0120only'", "severity": "ERROR"}
```

**Comment by schmmd** [comment]
OK--this is a nasty PR because it does a lot:

* Reverts serving the new coref model from a separate Dockerfile.  The approach was not going to scale.
* Updates many of our models to use a recent version of AllenNLP.
* Serves multiple static images for the first time, with fixes.
* Change how the demo is served, so there's one UI and each image is just an API.

After this is merged, I'd like to add some newer models.

**Comment by schmmd** [comment]
OK--merging down as I've tested it out in https://allennlp-demo.up.apps.allenai.org/.

**Pull request merged**